### PR TITLE
CMOS-69: Add Prometheus component docs

### PR DESCRIPTION
**Pull request by princerachit**
Signed-off-by: Prince Sinha <psinha36@gmail.com>

### CODE REVIEW

File: .aspell.en.pws [hunk 1]
```diff
@@ -59,3 +59,5 @@ CPUs
 MiB
 RBAC
 YAML
+de
```
Comment (patrick-stephens) [inline .aspell.en.pws:62]: I'm not a fan of this, I think it would be better to reword so we don't add an ignore for "de" and "facto" on their own.

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
+= Prometheus
+
+[abstract]
+The Couchbase Prometheus Server can be used to collect cloud-native metrics from Couchbase Servers.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:4]: I don't think you need to do cloud native here

Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:4]: We probably need to be more explicit that this is Prometheus in CMOS to distinguish the Prometheus instance in Couchbase Server 7+

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+https://prometheus.io/[Prometheus^] is a leading open-source monitoring solution. As a https://www.cncf.io/projects/prometheus/[graduated project^] of the Cloud Native Computing Foundation, it has become the de facto standard for metrics collection and alert generation on cloud platforms.
+
+Monitoring Couchbase is a critical aspect of production use case. Couchbase exposes detailed metrics that can be collected for visualization and other forms of monitoring. Prometheus fits perfectly as a monitoring solution for these exposed metrics to develop alerts for system issues, identify maintenance windows and other valuable business intelligence insights.
+
+image::component-prometheus.png[]
+
+== Get Started
+
+The Couchbase Prometheus Server is managed by the xref:architecture.adoc[Couchbase Monitoring Observability Stack]. Prometheus server is enabled by default as part of CMOS and it can be configured to monitor Couchbase cluster both for a Kubernetes based deployment done via Couchbase Operator Docs or for a Virtual Machine based deployment done via docker containers.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:16]: I think I'd reword this to be the CMOS managed Prometheus server.

Also it says "Couchbase Operator Docs", is that supposed to be a link to docs? 

This all works for non containers too.

File: docs/modules/ROOT/nav.adoc [hunk 4]
```diff
@@ -7,6 +7,9 @@
 * xref:cluster-monitor.adoc[Couchbase Cluster Monitor component]
 * xref:deployment-distributed.adoc[Distributed microservices deployment]
 
+.Components
+* xref:component-prometheus.adoc[Prometheus]
```
Comment (patrick-stephens) [inline docs/modules/ROOT/nav.adoc:11]: Are we going to link to this from a diagram or something? It doesn't seem to provide a huge level of value otherwise.

I'd suggest focusing on the Couchbase Server 7 improvements plus the cloud native existing support for Prometheus.

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
+= Prometheus
+
+[abstract]
+The Couchbase Prometheus Server can be used to collect cloud-native metrics from Couchbase Servers.
+
+== Overview
+
+https://prometheus.io/[Prometheus^] is a leading open-source monitoring solution. As a https://www.cncf.io/projects/prometheus/[graduated project^] of the Cloud Native Computing Foundation, it has become the de facto standard for metrics collection and alert generation on cloud platforms.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:8]: Replace de facto with industry

**Review by patrick-stephens — APPROVED** [review]

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 2 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-prometheus.adoc:4]: > We probably need to be more explicit that this is Prometheus in CMOS to distinguish the Prometheus instance in Couchbase Server 7+

From prometheus instances do you mean exporter here?

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 3 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-prometheus.adoc:16]: Do we have documentation for non container based deployment? I would rather not bring it up unless we have them.

File: docs/modules/ROOT/nav.adoc (hunk 4 above)
Comment (princerachit) [reply docs/modules/ROOT/nav.adoc:11]: I am not quite sure what you mean, the structure of pages is not clear to me. I am addressing [this ticket](https://issues.couchbase.com/browse/CMOS-69) for various components.

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 2 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-prometheus.adoc:4]: I would try to keep abstract minimal. Following sections explain things in more clarity.

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 3 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-prometheus.adoc:16]: >Also it says "Couchbase Operator Docs", is that supposed to be a link to docs?
Nice catch, will add link to operator page

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 2 above)
Comment (patrick-stephens) [reply docs/modules/ROOT/pages/component-prometheus.adoc]: No, Couchbase Server 7 provides its own Prometheus server internally which we can scrape - we do not need the exporter and the intention is to focus on this for 0.1.

Comment (patrick-stephens) [reply docs/modules/ROOT/pages/component-prometheus.adoc]: So my concern is that the "Couchbase Prometheus Server" is ambiguous - do you mean the one in server or the one in CMOS?

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 3 above)
Comment (patrick-stephens) [reply docs/modules/ROOT/pages/component-prometheus.adoc:16]: Ah I see what you're saying and it confused me so think it needs rewording slightly.

"or for a Virtual Machine based deployment done via docker containers" I thought you were talking about Couchbase Server here as you include it in the sentence about Couchbase Cluster which is the Server. Couchbase Server can be deployed via native binaries, containers and in Kubernetes. 

CMOS has to support all of those options and will be explicitly targeting the on-premise options (i.e. native and containers) for 0.1. CMOS itself will only support being deployed as a container.

I think this again is the confusion from the "Couchbase Prometheus Server", we need to be clear are we talking about CMOS or Server?

Additionally the inclusion of VM does not add anything here. You can do the same thing on a non-VM, by definition just a (bare metal) machine.

File: docs/modules/ROOT/nav.adoc (hunk 4 above)
Comment (patrick-stephens) [reply docs/modules/ROOT/nav.adoc:11]: Currently this page is just a bit of blurb about Prometheus which feels more like a Grafana press release than a Couchbase document. By all means we can have a little intro but it needs to focus on what it provides in the context of CMOS. 
We also want to highlight that CBS 7 has native Prometheus support too which we are aligning with and building on to provide a capable monitoring solution out of the box.

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 6]
```diff
@@ -0,0 +1,21 @@
+= Prometheus
+
+[abstract]
+The Couchbase Prometheus Server can be used to collect metrics from Couchbase Servers.
+
+== Overview
+
+https://prometheus.io/[Prometheus^] is a leading open-source monitoring solution. As a https://www.cncf.io/projects/prometheus/[graduated project^] of the Cloud Native Computing Foundation, it has become the industry standard for metrics collection and alert generation on cloud platforms.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:8]: See the other PR but all sentences should be on their own single line. It is much easier to review then and has no impact on the output.

**Review by patrick-stephens — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/nav.adoc (hunk 4 above)
Comment (patrick-stephens) [reply docs/modules/ROOT/nav.adoc:11]: The JIRA you linked includes scope for a diagram which was my original question: are we having a clickable diagram that then jumps to these pages? That might be a good idea and the landing pages can then call out the OSS docs.

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 3 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-prometheus.adoc:16]: So, does this sound better, `or for a on-premise deployment done via docker containers`? But again the deployment can be in cloud and not on premise.

Comment (patrick-stephens) [reply docs/modules/ROOT/pages/component-prometheus.adoc:16]: I think this section needs a bit of rewording, it's confusing CMOS deployment and Couchbase Server deployment.

It starts talking about CMOS enabled Prometheus then moves to Couchbase Cluster (i.e. Couchbase server) deployed via CAO or on-premise then goes back to talking about container deployment of CMOS all in the same sentence.

Something like:
"CMOS provides a Prometheus server internally that is enabled by default.
CMOS can be configured to scrape Couchbase clusters provided either as on-premise deployments (bare metal or VM) or Kubernetes deployments managed by the Couchbase Autonomous Operator.
CMOS itself is deployed as a container in both types of infrastructure, i.e. either running on a host as a manually orchestrated container or within Kubernetes as a pod."

Comment (patrick-stephens) [reply docs/modules/ROOT/pages/component-prometheus.adoc:16]: Explicitly we support Couchbase Server running as a native binary, i.e. not as a container. We also support it running as a container.

We only ever support CMOS running as a container for all use cases. This is what confused me initially and will confuse others so I think it needs to be clearer and shorter sentences.

File: docs/modules/ROOT/nav.adoc (hunk 4 above)
Comment (princerachit) [reply docs/modules/ROOT/nav.adoc:11]: I am not sure how this is to be structure tbh. Adding a dedicated section should appear itself in the navigation pane.

This section links to the actual page which has a diagram.

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 3 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-prometheus.adoc]: Since we are writing a page about cloud native database, I have tried to include some info about both Couchbase server prometheus and CMOS prometheus.

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 2 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-prometheus.adoc]: Updated info about both

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 3 above)
Comment (patrick-stephens) [reply docs/modules/ROOT/pages/component-prometheus.adoc]: That's fine just make it clear which is being used I think. As I say, I designed this stack and it confused me when reading it :-)

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+== Overview
+
+https://prometheus.io/[Prometheus^] is a leading open-source monitoring solution.
+As a https://www.cncf.io/projects/prometheus/[graduated project^] of the Cloud Native Computing Foundation, it has become the industry standard for metrics collection and alert generation on cloud platforms.
+
+Monitoring Couchbase is a critical aspect of production use case.
+Couchbase exposes detailed metrics via its own prometheus server starting with Couchbase Server 7+ that can be collected for visualization and other forms of monitoring.
+CMOS prometheus server can be used to scrape these metrics to develop alerts for system issues, identify maintenance windows and other valuable business intelligence insights.
+
+image::component-prometheus.png[]
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:15]: This will need the bugfix for Github rendering: 
```
ifdef::env-github[]
:imagesdir: https://github.com/couchbaselabs/observability/raw/main/docs/modules/ROOT/assets/images
endif::[]
```
Otherwise the images will not appear correctly in Github.

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+== Get Started
+
+The Couchbase Server Prometheus is shipped along with couchbase server 7+. The CMOS Prometheus Server is enabled by default in the xref:architecture.adoc[Couchbase Monitoring Observability Stack] container.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:19]: Split the lines please.

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+It can be configured to monitor Couchbase cluster by scraping metrics from the Couchbase server shipped prometheus endpoints.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:20]: We should be consistent with "Couchbase Server" - 'S' being uppercase.

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 10]
```diff
@@ -0,0 +1,25 @@
+= Prometheus
+
+[abstract]
+The Couchbase Prometheus Server is used to expose metrics and CMOS Prometheus server is used to scrape those metrics.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:4]: The CMOS Prometheus instance can scrape any Prometheus endpoint, e.g. we are looking to provide node exporter or other additional tooling. I think it might be better to reword just to make this distinction: CMOS provides a Prometheus Server instance that can scrape any supported endpoint, Couchbase Server 7+ provides an endpoint to scrape.

That way it fits with usage of Prometheus exporter for other versions of Couchbase Server.

**Review by patrick-stephens — CHANGES_REQUESTED** [review]
The diagram looks good but needs a slight tweak to use the term "Cluster Monitor" rather than "Cluster Manager". It would be good to add the Couchbase logo or similar to it to indicate it is (currently) proprietary.

**Comment by princerachit** [comment]
Can we get this merged now? I think page navigation and structuring can be addressed separately with Anil's help.

**Comment by patrick-stephens** [comment]
> Can we get this merged now? I think page navigation and structuring can be addressed separately with Anil's help.

Yeah I'll have a quick look shortly, apologies for the delay.

File: docs/modules/ROOT/pages/component-prometheus.adoc [hunk 11]
```diff
@@ -0,0 +1,21 @@
+= Prometheus
+
+[abstract]
+The Couchbase Prometheus Server can be used to collect cloud-native metrics from Couchbase Servers.
+
+== Overview
+
+https://prometheus.io/[Prometheus^] is a leading open-source monitoring solution. As a https://www.cncf.io/projects/prometheus/[graduated project^] of the Cloud Native Computing Foundation, it has become the industry standard for metrics collection and alert generation on cloud platforms.
```
Comment (patrick-stephens) [inline docs/modules/ROOT/pages/component-prometheus.adoc:8]: Separate lines please for each sentence then we're good.

**Comment by patrick-stephens** [comment]
> The diagram looks good but needs a slight tweak to use the term "Cluster Monitor" rather than "Cluster Manager". It would be good to add the Couchbase logo or similar to it to indicate it is (currently) proprietary.

Need to update the diagram as indicated, not "Cluster Manager" please but "Cluster Monitor".

File: docs/modules/ROOT/pages/component-prometheus.adoc (hunk 11 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/component-prometheus.adoc:8]: I wish there was a lint tool which could catch this

**Comment by princerachit** [comment]
My force pushed messed up the changes :(

**Comment by princerachit** [comment]
Here is the G-Doc to quickly turn my mess around https://docs.google.com/document/d/1wYKlRAR0kop4R3Uyk4k9BVnReqZOrLjrblOT5TxLsDE/edit?usp=sharing

**Comment by patrick-stephens** [comment]
OK updated with some comments. I'd be tempted just to close the PR and start a new one. Also @anilkumar29 probably should cover docs reviews now.

**Comment by princerachit** [comment]
That sounds like a good plan @patrick-stephens . Closing this PR
